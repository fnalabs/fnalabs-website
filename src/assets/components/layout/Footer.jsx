import React from 'react'
import { Link } from 'react-router-dom'

import { AnalyticsLink, Container, Tile } from 'common'
import { Email, FnALabsInverted, Github, NPM, Twitter } from 'icons'

export const Footer = () => (
  <footer className='footer is-dark'>
    <Container>
      <div className='columns'>
        <div className='column is-narrow content'>
          <p>Copyright &copy; 2018 FnA Labs</p>
          <p className='is-size-7'>Icons from <a href='https://fontawesome.com/'>Font Awesome</a> (<a href='https://fontawesome.com/license'>License</a>)</p>
          <p><a href='https://bulma.io'>
            <img src='https://fnalabs.github.io/fnalabs-assets/assets/made-with-bulma--dark.png' alt='Made with Bulma' width='128' height='24' />
          </a></p>
        </div>

        <div className='column'>
          <div className='columns is-mobile'>
            <nav className='column is-6 has-text-right content'>
              <p><strong>Sitemap</strong></p>
              <p><Link to='/'>Home</Link></p>
              <p><Link to='/projects'>Projects</Link></p>
              <p><Link to='/about'>About</Link></p>
              <p><Link to='/support'>Support</Link></p>
            </nav>

            <nav className='column is-6 has-text-left content'>
              <p><strong>Policies</strong></p>
              <p><Link to='/cookie'>Cookie</Link></p>
              <p><Link to='/privacy'>Privacy</Link></p>
            </nav>
          </div>
        </div>

        <div className='column is-narrow'>
          <nav className='level is-mobile'>
            <div className='level-item'>
              <AnalyticsLink to='https://github.com/fnalabs' target='_blank'>
                <span className='icon'>
                  <Github className='svg-link' />
                </span>
              </AnalyticsLink>
            </div>

            <div className='level-item'>
              <AnalyticsLink to='https://www.npmjs.com/org/fnalabs' target='_blank'>
                <span className='icon'>
                  <NPM className='svg-link' />
                </span>
              </AnalyticsLink>
            </div>

            <div className='level-item'>
              <AnalyticsLink to='https://twitter.com/FnA_Labs' target='_blank'>
                <span className='icon'>
                  <Twitter className='svg-link' />
                </span>
              </AnalyticsLink>
            </div>

            <div className='level-item'>
              <AnalyticsLink to='mailto:contact@fnalabs.com'>
                <span className='icon'>
                  <Email className='svg-link' />
                </span>
              </AnalyticsLink>
            </div>
          </nav>

          <Tile type='ancestor'>
            <Tile type='parent' vertical>
              <Tile type='child' centered><FnALabsInverted /></Tile>
              <Tile type='child' centered content>
                <p>Fn Awesome!</p>
              </Tile>
            </Tile>
          </Tile>
        </div>
      </div>
    </Container>
  </footer>
)
