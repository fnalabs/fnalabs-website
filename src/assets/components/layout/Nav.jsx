import React, { Component } from 'react'
import { withRouter } from 'react-router'
import { Link, NavLink } from 'react-router-dom'

import { Container } from 'common'
import { FnALabs } from 'icons'

class Nav extends Component {
  constructor (props) {
    super(props)

    this.state = {
      closed: true,
      location: ''
    }
  }

  handleToggleMenu = () => {
    this.setState({ closed: !this.state.closed })
  }

  static getDerivedStateFromProps (props, state) {
    if (props.location.pathname !== state.location) {
      return {
        closed: true,
        location: props.location.pathname
      }
    } else return null
  }

  render () {
    let navClasses = 'navbar is-spaced is-transparent is-absolute'
    if (this.state.location !== '/cookie/' && this.state.location !== '/privacy/') {
      navClasses += ' is-bold'
    }

    return (
      <nav className={navClasses} role='navigation' aria-label='main navigation'>
        <Container>
          <div className='navbar-brand'>
            <Link to='/' className='navbar-item'><FnALabs /></Link>

            <div className={`navbar-burger${this.state.closed ? '' : ' is-active'}`} onClick={this.handleToggleMenu}>
              <span />
              <span />
              <span />
            </div>
          </div>

          <div className={`navbar-menu${this.state.closed ? '' : ' is-active'}`}>
            <div className='navbar-end'>
              <NavLink to='/projects' activeClassName='is-active' className='navbar-item'>Projects</NavLink>
              <NavLink to='/about' activeClassName='is-active' className='navbar-item'>About</NavLink>
              <NavLink to='/support' activeClassName='is-active' className='navbar-item'>Support</NavLink>
            </div>
          </div>
        </Container>
      </nav>
    )
  }
}

export default withRouter(Nav)
