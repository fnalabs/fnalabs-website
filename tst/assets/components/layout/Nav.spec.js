/* eslint-env jest */
import React from 'react'
import { MemoryRouter } from 'react-router'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import { Nav } from 'layout'

describe('<Nav />', () => {
  it('should toggle mobile menu correctly', () => {
    const nav = mount(<MemoryRouter><Nav /></MemoryRouter>)

    expect(nav.find('.navbar-menu').hasClass('is-active')).toBe(false)
    nav.find('.navbar-burger').simulate('click')
    expect(nav.find('.navbar-menu').hasClass('is-active')).toBe(true)
    nav.find('.navbar-burger').simulate('click')
    expect(nav.find('.navbar-menu').hasClass('is-active')).toBe(false)
  })

  it('should render homepage nav correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/']}><Nav /></MemoryRouter>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render Projects page nav correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/projects']}><Nav /></MemoryRouter>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render About page nav correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/about']}><Nav /></MemoryRouter>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render Support page nav correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/support']}><Nav /></MemoryRouter>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render Cookie Policy page nav correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/cookie']}><Nav /></MemoryRouter>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render Privacy Policy page nav correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/privacy']}><Nav /></MemoryRouter>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
