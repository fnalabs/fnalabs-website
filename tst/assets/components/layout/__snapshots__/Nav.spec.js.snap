// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render About page nav correctly 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-spaced is-transparent is-absolute is-bold"
  role="navigation"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="/"
        onClick={[Function]}
      >
        <svg
          height="36"
          viewBox="0 0 290 180"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 37.143495,2.499996 H 158.81128 L 152.1301,36.609173 H 75.58958 L 69.26004,69.19457 h 72.08641 l -6.7984,34.10918 H 62.57886 L 48.16158,177.5 H 3.0343185 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 232.8253,145.61788 H 162.96596 L 145.61834,177.5 H 99.31894 L 198.24727,2.499996 h 51.57402 L 280.76569,177.5 h -42.9002 z m -52.86336,-31.7649 h 47.00268 L 216.41539,46.455121 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 201.26761,104 A 50.387058,50 0 0 1 150.88055,154 50.387058,50 0 0 1 100.4935,104 50.387058,50 0 0 1 150.88055,53.999997 50.387058,50 0 0 1 201.26761,104 Z"
            fill="#2e8b57"
          />
          <path
            d="m 131.63097,76.79366 h 15.67708 l 12.46875,37.69792 7.32813,-37.69792 h 13.27083 l -10.60937,54.43229 H 154.0893 l -12.46875,-37.69791 -7.29166,37.69791 H 121.0216 Z"
            fill="#fff"
            stroke="#fff"
            strokeWidth="5"
          />
        </svg>
      </a>
      <div
        className="navbar-burger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-end"
      >
        <a
          aria-current={null}
          className="navbar-item"
          href="/projects"
          onClick={[Function]}
        >
          Projects
        </a>
        <a
          aria-current="page"
          className="navbar-item is-active"
          href="/about"
          onClick={[Function]}
          style={Object {}}
        >
          About
        </a>
        <a
          aria-current={null}
          className="navbar-item"
          href="/support"
          onClick={[Function]}
        >
          Support
        </a>
      </div>
    </div>
  </div>
</nav>
`;

exports[`<Nav /> should render Cookie Policy page nav correctly 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-spaced is-transparent is-absolute is-bold"
  role="navigation"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="/"
        onClick={[Function]}
      >
        <svg
          height="36"
          viewBox="0 0 290 180"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 37.143495,2.499996 H 158.81128 L 152.1301,36.609173 H 75.58958 L 69.26004,69.19457 h 72.08641 l -6.7984,34.10918 H 62.57886 L 48.16158,177.5 H 3.0343185 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 232.8253,145.61788 H 162.96596 L 145.61834,177.5 H 99.31894 L 198.24727,2.499996 h 51.57402 L 280.76569,177.5 h -42.9002 z m -52.86336,-31.7649 h 47.00268 L 216.41539,46.455121 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 201.26761,104 A 50.387058,50 0 0 1 150.88055,154 50.387058,50 0 0 1 100.4935,104 50.387058,50 0 0 1 150.88055,53.999997 50.387058,50 0 0 1 201.26761,104 Z"
            fill="#2e8b57"
          />
          <path
            d="m 131.63097,76.79366 h 15.67708 l 12.46875,37.69792 7.32813,-37.69792 h 13.27083 l -10.60937,54.43229 H 154.0893 l -12.46875,-37.69791 -7.29166,37.69791 H 121.0216 Z"
            fill="#fff"
            stroke="#fff"
            strokeWidth="5"
          />
        </svg>
      </a>
      <div
        className="navbar-burger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-end"
      >
        <a
          aria-current={null}
          className="navbar-item"
          href="/projects"
          onClick={[Function]}
        >
          Projects
        </a>
        <a
          aria-current={null}
          className="navbar-item"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="navbar-item"
          href="/support"
          onClick={[Function]}
        >
          Support
        </a>
      </div>
    </div>
  </div>
</nav>
`;

exports[`<Nav /> should render Privacy Policy page nav correctly 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-spaced is-transparent is-absolute is-bold"
  role="navigation"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="/"
        onClick={[Function]}
      >
        <svg
          height="36"
          viewBox="0 0 290 180"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 37.143495,2.499996 H 158.81128 L 152.1301,36.609173 H 75.58958 L 69.26004,69.19457 h 72.08641 l -6.7984,34.10918 H 62.57886 L 48.16158,177.5 H 3.0343185 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 232.8253,145.61788 H 162.96596 L 145.61834,177.5 H 99.31894 L 198.24727,2.499996 h 51.57402 L 280.76569,177.5 h -42.9002 z m -52.86336,-31.7649 h 47.00268 L 216.41539,46.455121 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 201.26761,104 A 50.387058,50 0 0 1 150.88055,154 50.387058,50 0 0 1 100.4935,104 50.387058,50 0 0 1 150.88055,53.999997 50.387058,50 0 0 1 201.26761,104 Z"
            fill="#2e8b57"
          />
          <path
            d="m 131.63097,76.79366 h 15.67708 l 12.46875,37.69792 7.32813,-37.69792 h 13.27083 l -10.60937,54.43229 H 154.0893 l -12.46875,-37.69791 -7.29166,37.69791 H 121.0216 Z"
            fill="#fff"
            stroke="#fff"
            strokeWidth="5"
          />
        </svg>
      </a>
      <div
        className="navbar-burger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-end"
      >
        <a
          aria-current={null}
          className="navbar-item"
          href="/projects"
          onClick={[Function]}
        >
          Projects
        </a>
        <a
          aria-current={null}
          className="navbar-item"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="navbar-item"
          href="/support"
          onClick={[Function]}
        >
          Support
        </a>
      </div>
    </div>
  </div>
</nav>
`;

exports[`<Nav /> should render Projects page nav correctly 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-spaced is-transparent is-absolute is-bold"
  role="navigation"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="/"
        onClick={[Function]}
      >
        <svg
          height="36"
          viewBox="0 0 290 180"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 37.143495,2.499996 H 158.81128 L 152.1301,36.609173 H 75.58958 L 69.26004,69.19457 h 72.08641 l -6.7984,34.10918 H 62.57886 L 48.16158,177.5 H 3.0343185 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 232.8253,145.61788 H 162.96596 L 145.61834,177.5 H 99.31894 L 198.24727,2.499996 h 51.57402 L 280.76569,177.5 h -42.9002 z m -52.86336,-31.7649 h 47.00268 L 216.41539,46.455121 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 201.26761,104 A 50.387058,50 0 0 1 150.88055,154 50.387058,50 0 0 1 100.4935,104 50.387058,50 0 0 1 150.88055,53.999997 50.387058,50 0 0 1 201.26761,104 Z"
            fill="#2e8b57"
          />
          <path
            d="m 131.63097,76.79366 h 15.67708 l 12.46875,37.69792 7.32813,-37.69792 h 13.27083 l -10.60937,54.43229 H 154.0893 l -12.46875,-37.69791 -7.29166,37.69791 H 121.0216 Z"
            fill="#fff"
            stroke="#fff"
            strokeWidth="5"
          />
        </svg>
      </a>
      <div
        className="navbar-burger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-end"
      >
        <a
          aria-current="page"
          className="navbar-item is-active"
          href="/projects"
          onClick={[Function]}
          style={Object {}}
        >
          Projects
        </a>
        <a
          aria-current={null}
          className="navbar-item"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="navbar-item"
          href="/support"
          onClick={[Function]}
        >
          Support
        </a>
      </div>
    </div>
  </div>
</nav>
`;

exports[`<Nav /> should render Support page nav correctly 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-spaced is-transparent is-absolute is-bold"
  role="navigation"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="/"
        onClick={[Function]}
      >
        <svg
          height="36"
          viewBox="0 0 290 180"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 37.143495,2.499996 H 158.81128 L 152.1301,36.609173 H 75.58958 L 69.26004,69.19457 h 72.08641 l -6.7984,34.10918 H 62.57886 L 48.16158,177.5 H 3.0343185 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 232.8253,145.61788 H 162.96596 L 145.61834,177.5 H 99.31894 L 198.24727,2.499996 h 51.57402 L 280.76569,177.5 h -42.9002 z m -52.86336,-31.7649 h 47.00268 L 216.41539,46.455121 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 201.26761,104 A 50.387058,50 0 0 1 150.88055,154 50.387058,50 0 0 1 100.4935,104 50.387058,50 0 0 1 150.88055,53.999997 50.387058,50 0 0 1 201.26761,104 Z"
            fill="#2e8b57"
          />
          <path
            d="m 131.63097,76.79366 h 15.67708 l 12.46875,37.69792 7.32813,-37.69792 h 13.27083 l -10.60937,54.43229 H 154.0893 l -12.46875,-37.69791 -7.29166,37.69791 H 121.0216 Z"
            fill="#fff"
            stroke="#fff"
            strokeWidth="5"
          />
        </svg>
      </a>
      <div
        className="navbar-burger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-end"
      >
        <a
          aria-current={null}
          className="navbar-item"
          href="/projects"
          onClick={[Function]}
        >
          Projects
        </a>
        <a
          aria-current={null}
          className="navbar-item"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current="page"
          className="navbar-item is-active"
          href="/support"
          onClick={[Function]}
          style={Object {}}
        >
          Support
        </a>
      </div>
    </div>
  </div>
</nav>
`;

exports[`<Nav /> should render homepage nav correctly 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-spaced is-transparent is-absolute is-bold"
  role="navigation"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="/"
        onClick={[Function]}
      >
        <svg
          height="36"
          viewBox="0 0 290 180"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 37.143495,2.499996 H 158.81128 L 152.1301,36.609173 H 75.58958 L 69.26004,69.19457 h 72.08641 l -6.7984,34.10918 H 62.57886 L 48.16158,177.5 H 3.0343185 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 232.8253,145.61788 H 162.96596 L 145.61834,177.5 H 99.31894 L 198.24727,2.499996 h 51.57402 L 280.76569,177.5 h -42.9002 z m -52.86336,-31.7649 h 47.00268 L 216.41539,46.455121 Z"
            fill="#00008b"
            stroke="#00008b"
            strokeWidth="5"
          />
          <path
            d="M 201.26761,104 A 50.387058,50 0 0 1 150.88055,154 50.387058,50 0 0 1 100.4935,104 50.387058,50 0 0 1 150.88055,53.999997 50.387058,50 0 0 1 201.26761,104 Z"
            fill="#2e8b57"
          />
          <path
            d="m 131.63097,76.79366 h 15.67708 l 12.46875,37.69792 7.32813,-37.69792 h 13.27083 l -10.60937,54.43229 H 154.0893 l -12.46875,-37.69791 -7.29166,37.69791 H 121.0216 Z"
            fill="#fff"
            stroke="#fff"
            strokeWidth="5"
          />
        </svg>
      </a>
      <div
        className="navbar-burger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-end"
      >
        <a
          aria-current={null}
          className="navbar-item"
          href="/projects"
          onClick={[Function]}
        >
          Projects
        </a>
        <a
          aria-current={null}
          className="navbar-item"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="navbar-item"
          href="/support"
          onClick={[Function]}
        >
          Support
        </a>
      </div>
    </div>
  </div>
</nav>
`;
